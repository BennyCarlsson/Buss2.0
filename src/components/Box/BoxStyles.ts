import styled from "styled-components"

export const Box = styled.div`
  border: 2px solid #a3a3a3;
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 20vh;
  border-radius: 20px;
`
