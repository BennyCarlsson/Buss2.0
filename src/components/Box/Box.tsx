import React from "react"
import { Box as StyledBox } from "./BoxStyles"

const Box: React.FC = (props) => {
  return <StyledBox>{props.children}</StyledBox>
}

export default Box
