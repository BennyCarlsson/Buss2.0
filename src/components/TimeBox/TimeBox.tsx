import React from "react"
import Box from "components/Box/Box"
interface Props {
  time: string
}

const TimeBox: React.FC<Props> = (props) => {
  return <Box>{props.time}</Box>
}

export default TimeBox
