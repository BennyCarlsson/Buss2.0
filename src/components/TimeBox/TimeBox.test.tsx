import React from "react"
import { shallow } from "enzyme"
import renderer from "react-test-renderer"
import Box from "../Box/Box"
import TimeBox from "./TimeBox"

describe("Snapshot Tests", () => {
  it("render TimeBox component", () => {
    const tree = renderer.create(<TimeBox time="11:12"/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe("TimeBox", () => {
  it("render TimeBox", () => {
    const wrapper = shallow(<TimeBox time="12:12" />)
    expect(wrapper.contains(<Box><p>12:12</p></Box>)).toEqual(true)
  })
})
