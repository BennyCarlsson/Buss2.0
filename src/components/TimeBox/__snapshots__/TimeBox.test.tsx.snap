// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Tests render TimeBox component 1`] = `
<div
  className="sc-bdnylx boKVLG"
>
  11:12
</div>
`;
